packages/wallet/test/index.spec.ts

```diff
@@ -27,617 +27,659 @@ const isRunningInKarma = (): boolean => {
+    const selectors = permuIdx
+      .toString(radix)
+      .padStart(keys.length, '0')
+      .split('')
+      .map((v) => parseInt(v, 10))
     const obj: any = {}
-    Object.entries(opts).forEach(([key, val]: [string, string | Uint8Array]) => {
-      obj[key] = typeof val === 'string' ? '0x' + val : val
+    zip(selectors, keys).forEach(([sel, k]: [number, string]) => {
+      if ((objs as any)[sel].hasOwnProperty(k) === true) {
+        obj[k] = (objs as any)[sel][k]
+      }
     })
-    return obj
+    permus.push(obj)
   }
+  return permus
+}
 
-  let permutations = makePermutations(strKdfOptions, buffKdfOptions)
+const makeEthersOptions = (opts: object) => {
+  const obj: any = {}
+  Object.entries(opts).forEach(([key, val]: [string, string | Uint8Array]) => {
+    obj[key] = typeof val === 'string' ? '0x' + val : val
+  })
+  return obj
+}
 
-  if (isRunningInKarma()) {
-    // These tests take a long time in the browser due to
-    // the amount of permutations so we will shorten them.
-    permutations = permutations.slice(1)
-  }
+let permutations = makePermutations(strKdfOptions, buffKdfOptions)
 
-  it('should work with PBKDF2', async function () {
-    this.timeout(0) // never
-    const w =
-      '{"version":3,"id":"7e59dc02-8d42-409d-b29a-a8a0f862cc81","address":"b14ab53e38da1c172f877dbc6d65e4a1b0474c3c","crypto":{"ciphertext":"01ee7f1a3c8d187ea244c92eea9e332ab0bb2b4c902d89bdd71f80dc384da1be","cipherparams":{"iv":"cecacd85e9cb89788b5aab2f93361233"},"cipher":"aes-128-ctr","kdf":"pbkdf2","kdfparams":{"dklen":32,"salt":"dc9e4a98886738bd8aae134a1f89aaa5a502c3fbd10e336136d4d5fe47448ad6","c":262144,"prf":"hmac-sha256"},"mac":"0c02cd0badfebd5e783e0cf41448f84086a96365fc3456716c33641a86ebc7cc"}}'
-
-    await Promise.all(
-      permutations.map(async function ({ salt, iv, uuid }) {
-        const encFixtureWallet = await fixtureWallet.toV3String(pw, {
-          kdf: 'pbkdf2',
-          c: n,
-          uuid,
-          salt,
-          iv,
-        })
+if (isRunningInKarma()) {
+  // These tests take a long time in the browser due to
+  // the amount of permutations so we will shorten them.
+  permutations = permutations.slice(1)
+}
+
+tape('.toV3(): should work with PBKDF2', async (t) => {
+  const w =
+    '{"version":3,"id":"7e59dc02-8d42-409d-b29a-a8a0f862cc81","address":"b14ab53e38da1c172f877dbc6d65e4a1b0474c3c","crypto":{"ciphertext":"01ee7f1a3c8d187ea244c92eea9e332ab0bb2b4c902d89bdd71f80dc384da1be","cipherparams":{"iv":"cecacd85e9cb89788b5aab2f93361233"},"cipher":"aes-128-ctr","kdf":"pbkdf2","kdfparams":{"dklen":32,"salt":"dc9e4a98886738bd8aae134a1f89aaa5a502c3fbd10e336136d4d5fe47448ad6","c":262144,"prf":"hmac-sha256"},"mac":"0c02cd0badfebd5e783e0cf41448f84086a96365fc3456716c33641a86ebc7cc"}}'
+
+  for (const perm of permutations) {
+    const encFixtureWallet = await fixtureWallet.toV3String(pw, {
+      kdf: 'pbkdf2',
+      c: n,
+      uuid: perm.uuid,
+      salt: perm.salt,
+      iv: perm.iv,
+    })
```

Comment:
I think it might be more efficient to keep the `Promise.all` rather than looping through promises, since with Promise.all all promises will be started simultaneously, while here they will be run sequentially.

Reply:
Hmm, ok, let me look at it with it changed back.  This whole section of the tests was almost incomprehensible to me until I unwound everything.